src/main/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileDeleter.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.reform.userprofileapi.service.impl;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.userprofileapi.controller.response.UserProfilesDeletionResponse;
+import uk.gov.hmcts.reform.userprofileapi.domain.entities.UserProfile;
+import uk.gov.hmcts.reform.userprofileapi.exception.ResourceNotFoundException;
+import uk.gov.hmcts.reform.userprofileapi.repository.UserProfileRepository;
+import uk.gov.hmcts.reform.userprofileapi.resource.UserProfilesDeletionData;
+import uk.gov.hmcts.reform.userprofileapi.service.AuditService;
+import uk.gov.hmcts.reform.userprofileapi.service.ResourceDeleter;
+
+@Service
+@Slf4j
+public class UserProfileDeleter implements ResourceDeleter<UserProfilesDeletionData> {
+
+    @Autowired
+    private UserProfileRepository userProfileRepository;
+
+    @Autowired
+    private AuditService auditService;
+
+    @Override
+    @Transactional
+    public UserProfilesDeletionResponse delete(UserProfilesDeletionData profilesData) {
+
+        List<UserProfile> userProfiles = new ArrayList<UserProfile>();
+        Set<String> userIds = new HashSet<String>(profilesData.getUserIds());
+        userIds.forEach(userId -> userProfiles.add(validateUserStatus(userId)));
+
+        return deleteUserProfiles(userProfiles);
+    }
+
+    /**
+     * Either delete all the audit and userProfiles data from the data base or none.
+     *
+     */
+    private UserProfilesDeletionResponse deleteUserProfiles(List<UserProfile> userProfiles) {
+
+        userProfileRepository.deleteAll(userProfiles);
+        UserProfilesDeletionResponse deletionResponse = new UserProfilesDeletionResponse();
+        HttpStatus status = HttpStatus.NO_CONTENT;
+        deletionResponse.setMessage("UserProfiles Successfully Deleted");
+        deletionResponse.setStatusCode(status.value());
+        auditService.persistAudit(deletionResponse);
+        return deletionResponse;
+
+    }
+
+    /**
+     * This method is used to find the user profile exist or not with the give userId.
+     *
+     */
+    private UserProfile validateUserStatus(String userId) {
+        Optional<UserProfile> userProfileOptional = userProfileRepository.findByIdamId(userId.trim());
+        if (!userProfileOptional.isPresent()) {
+            throw new ResourceNotFoundException("could not find user profile for userId:" + userId);
```

Comment:
since you are throwing exception from here.. it will rollback the whole transaction in multiple user scenario..(When one of user not found).. what is the error message bein returned in this case?
there should be an integration and functional test to test this scenario

Reply:
it fails while collecting the user profiles itself and throws  404 status code with could not find user profile for userId: userId . once all user profiles collected successfully then it goes to delete the user profiles.